shopify/resources/price_rule.py

```diff
@@ -0,0 +1,10 @@
+from ..base import ShopifyResource
+from .discount_code import DiscountCode
+
+class PriceRule(ShopifyResource):
+    def add_discount(self, discount_code = DiscountCode()):
```

Comment:
Also here the discount_code will always be the same instance since the default param is evaluated once

```python
class Bar:
  pass

def test(foo = Bar()):
  print(foo)
  
test()  # <__builtin__.Bar instance at 0x7f74ecb1f170>
test()  # <__builtin__.Bar instance at 0x7f74ecb1f170>
```